### ISSUE TITLE
Limit the allowed roles for  aria-haspopup   

### ISSUE THREAD

**Issue by shirsha** [w3c/aria#1747]
# Describe your concern
As per the aria  specification, aria-haspopup can be used on  links, textbox, tabs, slider.
Links are used for navigation. So, what is use of providing aria-haspopup for links?
This will create confusion to the screen reader users as it will be narrated as menu item.

I have same question for textbox, tabs and slider. If we add aria-popup="true" to these roles they will be announced as menu button which will be misleading

# Link to the version of the specification or documentation you were looking at at.

Link to documentation: https://w3c.github.io/aria/#aria-haspopup,

Does the issue exists in [the editors draft](https://w3c.github.io/aria/) (the editors draft is the most recent draft of the specification)?

For more discussion please see https://web-a11y.slack.com/archives/C042TSFGN/p1653319076297029

**Comment by scottaohara** [comment]
`aria-haspopup` was already significantly scaled back on what roles allow it during aria 1.2.

we have the existing issue of [Clarify usage of aria-haspopup](https://github.com/w3c/aria/issues/1024), which Matt and I are assigned to work on, and which would cover remaining use cases as we look to re-align on what the attribute should be used for.

Per your assertions of a link with aria-haspopup saying "...it will be narrated as menu item" and "textbox, tabs and slider. If we add aria-popup="true" to these roles they will be announced as menu button"   Do you have examples to repro this?  While what you're saying is true if using `aria-haspopup` on a button, other elements did not have their role changed to "menu button" in the past, nor does a quick testing in firefox chrome and safari seem to yield a role change for at least link and textbox.  If you have data to show otherwise, it'd be great to see it so bugs can be filed.

**Comment by jnurthen** [comment]
@scottaohara @shirsha is this still an issue or is this covered by the related issues. Can this be closed?

**Comment by scottaohara** [comment]
I think it should be closed.
